protostar/migrator/cheatcodes/migrator_deploy_contract_cheatcode.py

```diff
@@ -1,32 +1,59 @@
 import asyncio
+import json
 from dataclasses import dataclass
-from logging import getLogger
-from typing import Any, Optional
+from pathlib import Path
+from typing import Any, Optional, Union
 
+from starknet_py.net.models import parse_address
+from starknet_py.net.signer import BaseSigner
 from typing_extensions import Protocol
 
+from protostar.migrator.cheatcodes.migrator_declare_cheatcode import (
+    DeclareCheatcodeNetworkConfig,
+    ValidatedDeclareCheatcodeNetworkConfig,
+)
 from protostar.migrator.migrator_contract_identifier_resolver import (
     MigratorContractIdentifierResolver,
 )
-from protostar.starknet import Cheatcode, KeywordOnlyArgumentCheatcodeException
-from protostar.starknet_gateway.gateway_facade import GatewayFacade
+from protostar.starknet import (
+    Cheatcode,
+    KeywordOnlyArgumentCheatcodeException,
+    CheatcodeException,
+)
+from protostar.starknet_gateway.gateway_facade import (
+    GatewayFacade,
+    InputValidationException,
+)
 from protostar.starknet.data_transformer import CairoOrPythonData
 
-from .network_config import CheatcodeNetworkConfig, ValidatedCheatcodeNetworkConfig
+
+class DeployCheatcodeNetworkConfig(DeclareCheatcodeNetworkConfig):
+    pass
+
+
+@dataclass
+class ValidatedDeployCheatcodeNetworkConfig(ValidatedDeclareCheatcodeNetworkConfig):
+    @classmethod
+    def from_deploy_cheatcode_network_config(
+        cls, config: Optional[DeployCheatcodeNetworkConfig]
+    ) -> "ValidatedDeclareCheatcodeNetworkConfig":
+        return super().from_declare_cheatcode_network_config(config)
 
 
 @dataclass(frozen=True)
 class DeployedContract:
     contract_address: int
 
 
-class DeployContractCheatcodeProtocol(Protocol):
+class MigratorDeployContractCheatcodeProtocol(Protocol):
     # pylint bug ?
     # pylint: disable=keyword-arg-before-vararg
     def __call__(
         self,
-        contract: str,
+        class_hash: Union[str, int],
         constructor_args: Optional[CairoOrPythonData] = None,
+        abi_path: Optional[str] = None,
+        salt: Optional[int] = None,
         *args: Any,
         config: Optional[Any] = None,
```

Comment:
Some arguments are passed directly and some through config. We don't need to maintain a common interface for `deploy_contract` cheatcodes and since this is a breaking change, config can be removed.

Reply:
> The thing being that we can mantain a common interface without a huge hassle, is the reason we should keep the interface.

This PR breaks the common interface (between testing and migrator `deploy_contract`). The testing environment isn't going to use UDC. 

I'd update the docs in this PR to bring the repo from one consistent state to another.

Reply:
Merging into a side branch, so will update the docs in the next one.